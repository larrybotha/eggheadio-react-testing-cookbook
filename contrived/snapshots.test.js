import React from 'react'
import renderer from 'react-test-renderer'
import {makeSomeRequestAndGetSomeResponse, getFilteredStuff, MyComponent} from './snapshots'

test('objects', async () => {
  const httpResponse = await makeSomeRequestAndGetSomeResponse()
  expect(httpResponse).toMatchSnapshot()
})

test('arrays', () => {
  const someFilteredList = getFilteredStuff('filter')
  expect(someFilteredList).toMatchSnapshot()
})

test('jsx', () => {
  const component = renderer.create(<MyComponent name="John" />)
  expect(component).toMatchSnapshot()
})
