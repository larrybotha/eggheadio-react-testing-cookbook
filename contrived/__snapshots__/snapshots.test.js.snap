// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrays 1`] = `
Array [
  "Vacuums have filters",
  "I will not be filtered out",
]
`;

exports[`jsx 1`] = `
<div>
  <h1>
    Hello
  </h1>
   there 
  <strong>
    John!
  </strong>
</div>
`;

exports[`objects 1`] = `
Object {
  "age": 32,
  "email": "123@example.com",
  "foo": "bar",
  "id": "3423490",
  "name": "Joe",
}
`;
